src/commands/upload_file.ts

```diff
@@ -99,27 +108,42 @@ new CLICommand({
 				feeMultiple: options.boost as FeeMultiple,
 				dryRun: options.dryRun
 			});
+
 			await Promise.all(
 				filesToUpload.map(async (fileToUpload) => {
-					if (!fileToUpload.parentFolderId || !fileToUpload.localFilePath) {
-						console.log(`Bad file: ${JSON.stringify(fileToUpload)}`);
-						process.exit(1);
-					}
-					const { parentFolderId, localFilePath, destinationFileName } = options;
+					const { parentFolderId, wrappedEntity, destinationFileName } = fileToUpload;
 
 					const result = await (async () => {
 						if (await parameters.getIsPrivate()) {
 							const driveId = await arDrive.getDriveIdForFolderId(parentFolderId);
 							const driveKey = await parameters.getDriveKey(driveId);
 
-							return arDrive.uploadPrivateFile(
-								parentFolderId,
-								localFilePath,
-								driveKey,
-								destinationFileName
-							);
+							if (isFolder(wrappedEntity)) {
+								return arDrive.createPrivateFolderAndUploadChildren(
+									parentFolderId,
+									wrappedEntity,
+									driveKey,
+									destinationFileName
+								);
+							} else {
+								wrappedEntity;
```

Comment:
Straggler?

Reply:
Nice hiding spot, cleaned up in this commit a55726c